types/react-router/index.d.ts

```diff
@@ -61,7 +61,7 @@ declare module 'react-router' {
 
   interface RouteProps {
     location?: H.Location;
-    component?: React.SFC<RouteComponentProps<any> | void | {}> | React.ComponentClass<RouteComponentProps<any> | void | {}>;
```

Comment:
Why did you removed `RouteComponentProps`?
Also I believe in the original code it should have been `React.SFC<RouteComponentProps<any> | {}>` instead of `React.SFC<RouteComponentProps<any> | void>`, see https://github.com/Microsoft/TypeScript/issues/15409
No need for `void | {}` I think.